### PR TITLE
Added read string ability

### PR DESCRIPTION
**Pull request by kkoser**
This commit adds a new function called keyboard_read_str, which reads
in a string from the keyboard until a return character is encountered.
It echos all input to the console as well, so the user can see what
he or she is typing. In the future, this can be made optional.

Tested in main by echoing and during comparisons with strings.

### CODE REVIEW

File: src/keyboard.c [hunk 1]
```diff
@@ -131,6 +133,25 @@ char keyboard_read()
...
+	char c = keyboard_read();
+	while(c != '\n' && c != '\r' && i < BUFFER_SIZE) {
+		console_putchar(c);
+		str_buffer[i++] = c;
+		
+		c = keyboard_read();
+	}
+	
+	// Mark the end of the line
+	str_buffer[i] = '\0';
```
Comment (alexmwu) [inline src/keyboard.c:147]: breaks array bounds when i = 256

Comment (kkoser) [inline src/keyboard.c:147]: Fixed!

**Comment by rboccabe** [comment]
const char \* for the return value?
I don't think we want to give anyone the ability to overwrite the keyboard buffer, even if once the line is read in we don't use the buffer again.

**Comment by jhamilt5** [comment]
Looks good

**Pull request merged**